pkg/auth/handler/update_metadata.go

```diff
@@ -125,11 +125,11 @@ func (h UpdateMetadataHandler) mergeMetadata(
 	loginIDMetadataKeys := h.PasswordAuthProvider.GetLoginIDMetadataFlattenedKeys()
 
 	output := make(map[string]interface{})
-	for k := range orgMetadata {
-		output[k] = orgMetadata[k]
-	}
 	for k, v := range inMetadata {
-		if !utils.StringSliceContains(loginIDMetadataKeys, k) {
+		output[k] = v
+	}
+	for k, v := range orgMetadata {
+		if utils.StringSliceContains(loginIDMetadataKeys, k) {
```

Comment:
Looks weird, why update with old data after new data?

I thought the previous implementation is able to delete attribute by setting value to `null`? This is the same behaviour as v1 record save. I think you may write a more complete test case, deleting non login id keys.

Reply:
My original intention is to copy all attributes from new data and overwrite the original loginID value.

I update the implementation as:

```
	output := make(map[string]interface{})
	// copy loginID to the profile
	for _, v := range loginIDMetadataKeys {
		if pv, ok := orgMetadata[v]; ok {
			output[v] = pv
		}
	}
	// copy rest attributes to the profile
	for k, v := range inMetadata {
		if !utils.StringSliceContains(loginIDMetadataKeys, k) {
			output[k] = v
		}
	}
```

Feel free to Let me know your comments about this. Thanks.